File: two_sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        index = dict()

        for i, num in enumerate(nums):
            if (target - num) in index:
                return [i, index[target - num]]
            index[num] = i
        return []

File: two_sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
	const index = {};
	var result = [];
	nums.forEach((num, i) => {
		if (target - num in index) {
			result = [i, index[target - num]];
		} else {
			index[num] = i;
		}
	});
	return result;
};
